tests/neg-custom-args/captures/i21868.scala

```diff
@@ -0,0 +1,14 @@
+import caps.*
+
+trait AbstractWrong:
+    type C <: CapSet
+    def boom(): Unit^{C^} // error
+
+trait Abstract:
+    type C <: CapSet^
```

Comment:
Idealy, we'd like to retrict the capture parameter and member bounded by `>: CapSet^{} <: CapSet^`, which can forbid instantiating to meaningless types, like `Nothing`.

Reply:
That's certainly cleaner. The question is how far we want to take it for now. A fully general solution would also have to deal with the presence of explicit lower bounds, which could be themselves type variables, etc.
```scala
trait Abstract[L >: CapSet^{}]:
    type T >: L <: CapSet^
```
Maybe let's track this in a separate issue? The point of the current fix is to prevent the uncaught exception and compiler crash.

Reply:
As the below examples show, there's a still a few quirks. I'm also somewhat uneasy with automatically presuming that the lower bound of a `CapSet^`-bounded type variable is `CapSet^{}`. E.g., should this always be the case, or only with cc enabled? 

Since the specific issues are resolved, let's merge this and have a separate issue to make capture set intervals and subtyping airtight.